crates/apollo-compiler/src/validation/field.rs

```diff
@@ -10,11 +11,41 @@ use crate::{
 pub fn validate_field(
     db: &dyn ValidationDatabase,
     field: Arc<hir::Field>,
+    type_def: hir::TypeDefinition,
```

Comment:
I am wondering if it's necessary to have a type_def param? We can always get a field's type and it's type definition:
 
```rust
let ty = field.ty(db.upcast());
let type_def = field_type.type_def(db.upcast());
```

or is there something else you would like to do with this `TypeDefinition`?

Reply:
The validation error for a missing field has a message along the lines of: `Field name is not defined on Type Person`. 

For this, I need the type def or name of the field's _parent type_ (Person). `field.ty` would return the name of the _field type_ (`None`, or `String` in case the field exists).
`field.parent_obj` would be the perfect choice, but it is optional. 
I just double-checked and it seems like we can still use it since it should be `Some` in case the field is defined on a valid parent object. This is ensured by performing the valid root operation type check first, to verify that the parent exists (explanation your review comment above)

> ordering shouldn't really matter in validation 🤔 why does this need to be after root operation type check?

If the paren't doesn't exist, we wouldn't continue validation and simply output the diagnostic.

Reply:
>For this, I need the type def or name of the field's parent type (Person). field.ty would return the name of the field type (None, or String in case the field exists).
ah, gotcha! let's call it `parent_ty` to disambiguate then. But I see you found a different way about it, which I think works well too.

> This is ensured by performing the valid root operation type check first, to verify that the parent exists (explanation your review comment above)

Ah! I am sorry, I didn't quite follow the first time around.